sql/parser/builtins.go

```diff
@@ -62,103 +80,94 @@ type builtin struct {
 // Need to figure out if there is a standard set of functions or if we can
 // customize this to our liking. Would be good to support type conversion
 // functions.
-var builtins = map[string]builtin{
+var builtins = map[string][]builtin{
 	// TODO(XisiHuang): support encoding, i.e., length(str, encoding).
-	"length": stringBuiltin(func(s string) (Datum, error) {
+	"length": {stringBuiltin(func(s string) (Datum, error) {
 		return DInt(len(s)), nil
-	}),
+	})},
 
-	"lower": stringBuiltin(func(s string) (Datum, error) {
+	"lower": {stringBuiltin(func(s string) (Datum, error) {
 		return DString(strings.ToLower(s)), nil
-	}),
+	})},
 
-	"upper": stringBuiltin(func(s string) (Datum, error) {
+	"upper": {stringBuiltin(func(s string) (Datum, error) {
 		return DString(strings.ToUpper(s)), nil
-	}),
+	})},
 
-	// TODO(XisiHuang): support the substring(str FROM x FOR y) syntax.
-	"substring": {
-		nArgs: -1,
+	"substr":    substringImpls,
+	"substring": substringImpls,
+
+	// concat concatenates the text representations of all the arguments.
+	// NULL arguments are ignored.
+	"concat": {
+		builtin{
```

Comment:
Do you need the `builtin` name here? Below in `substringImpls` it looks like the compiler can intuit the name.